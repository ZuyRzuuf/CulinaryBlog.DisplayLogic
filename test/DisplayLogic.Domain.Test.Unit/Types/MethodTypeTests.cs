using DisplayLogic.Domain.Types;
using HotChocolate;
using HotChocolate.Types;
using Microsoft.Extensions.DependencyInjection;
using Snapshooter.Xunit;

namespace DisplayLogic.Domain.Test.Unit.Types;

public class MethodTypeTests
{
    [Fact]
    public void MethodType_Schema_Snapshot()
    {
        // Arrange
        var services = new ServiceCollection()
            .AddGraphQLServer()
            .AddQueryType<QueryType>()
            .AddType<MethodType>()
            .Services
            .BuildServiceProvider();

        var schema = SchemaBuilder.New()
            .AddServices(services)
            .AddQueryType<QueryType>()
            .AddType<MethodType>()
            .Create();

        // Assert
        schema.ToString().MatchSnapshot();
    }

    // Add ArticleType_Fields_ExistAndAreCorrectType
    [Fact]
    public void MethodType_Fields_ExistAndAreCorrectType()
    {
        // Arrange
        var schema = SchemaBuilder.New()
            .AddQueryType<QueryType>()
            .AddType<MethodType>()
            .Create();

        // Act
        var fields = schema.GetType<ObjectType>("Method").Fields;

        // Assert
        Assert.Contains(fields, x => x.Name == "id" && x.Type.NamedType().Name.Equals("UUID"));
        Assert.Contains(fields, x => x.Name == "name" && x.Type.NamedType().Name.Equals("String"));
    }
}