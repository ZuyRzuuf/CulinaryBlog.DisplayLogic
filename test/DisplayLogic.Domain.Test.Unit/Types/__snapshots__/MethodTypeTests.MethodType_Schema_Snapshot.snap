schema {
  query: Query
}

type Article {
  id: UUID!
  title: String!
  content: String!
  author: Author!
  publishedDate: DateTime!
  imageUrl: String!
  tags: [Tag!]!
  comments: [Comment!]!
}

type Author {
  id: UUID!
  username: String!
}

type BlogData {
  articles(filters: ArticleFilter): [Article]
  recipes: [Recipe]
  article(articleId: ID!): Article
  recipe(recipeId: ID!): Recipe
}

type Category {
  id: UUID!
  name: String!
}

type Comment {
  id: UUID!
  content: String!
  author: Author!
  createdAt: DateTime!
  articleId: UUID!
  recipeId: UUID!
}

type Cuisine {
  id: UUID!
  name: String!
}

type Diet {
  id: UUID!
  name: String!
}

type Ingredient {
  id: UUID!
  name: String!
}

type Method {
  id: UUID!
  name: String!
}

type Query {
  articles: [Article]
  recipes: [Recipe]
  blogData(articleId: UUID = "00000000-0000-0000-0000-000000000000" recipeId: UUID = "00000000-0000-0000-0000-000000000000"): BlogData
  articleById(id: UUID!): Article!
}

type Recipe {
  id: UUID!
  title: String!
  ingredients: [Ingredient!]!
  instructions: [String!]!
  author: Author!
  publishedDate: DateTime!
  imageUrl: String!
  category: Category!
  cuisine: Cuisine!
  diet: Diet!
  method: Method!
  season: Season!
  tags: [Tag!]!
  comments: [Comment!]!
}

type Season {
  id: UUID!
  name: String!
}

type Tag {
  id: UUID!
  name: String!
}

input ArticleFilter {
  articleIds: [UUID]
  tagIds: [UUID]
  tagNames: [String]
  articleId: UUID
}

"The `@specifiedBy` directive is used within the type system definition language to provide a URL for specifying the behavior of custom scalar definitions."
directive @specifiedBy("The specifiedBy URL points to a human-readable specification. This field will only read a result for scalar types." url: String!) on SCALAR

"The `DateTime` scalar represents an ISO-8601 compliant date time type."
scalar DateTime @specifiedBy(url: "https:\/\/www.graphql-scalars.com\/date-time")

scalar UUID @specifiedBy(url: "https:\/\/tools.ietf.org\/html\/rfc4122")
